src/common/stattextcmn.cpp

```diff
@@ -213,8 +213,46 @@ class wxLabelWrapper : public wxTextWrapper
 
 void wxStaticTextBase::Wrap(int width)
 {
+    if (width == m_currentWrap) return;
+    m_currentWrap = width;
+
+    // Allow for repeated calls to Wrap with different values
+    if (m_unwrappedLabel.IsNull()) 
+    {
+        m_unwrappedLabel = GetLabel();
+    } 
+    else 
+    {
+        SetLabel( m_unwrappedLabel );
+    }
     wxLabelWrapper wrapper;
     wrapper.WrapLabel(this, width);
+    InvalidateBestSize();
+}
+
+wxSize wxStaticTextBase::GetEffectiveMinSizeFirstPass() const
+{
+    // While wxWrapSizer can only wrap entire controls, a text paragraph
+    // could theoretically wrap at a few letters, so we start with
+    // requesting very little space in the first pass
+    return wxSize( 10, 10 );
+}
+
+bool wxStaticTextBase::InformFirstDirection(int direction, int size, int WXUNUSED(availableOtherDir))
+{
+    // In the second pass, this control has been given "size" amount of
+    // space in the horizontal direction. Wrap there and report a new
+    // GetEffectiveMinSize() from then on.
+
+    if (direction != wxHORIZONTAL) return false;
+
+    int style = GetWindowStyleFlag();
+    SetWindowStyleFlag( style | wxST_NO_AUTORESIZE );
```

Comment:
It looks like we could avoid calling it unnecessarily (and also below):

```suggestion
    if ( !(style & wxST_NO_AUTORESIZE) )
        SetWindowStyleFlag( style | wxST_NO_AUTORESIZE );
```